presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -105,7 +109,41 @@ public Footer readFooter(HiveWriterVersion hiveWriterVersion, InputStream inputS
                 toStripeInformation(footer.getStripesList()),
                 toType(footer.getTypesList()),
                 fileStats,
-                toUserMetadata(footer.getMetadataList()));
+                toUserMetadata(footer.getMetadataList()),
+                toEncryption(footer.getEncryption()));
+    }
+
+    private Optional<Encryption> toEncryption(DwrfProto.Encryption encryption)
+    {
+        if (encryption.getEncryptionGroupsCount() == 0) {
```

Comment:
the contract is that proto.encryption is optional. If it exists, it should have at least one encryption group